### PR TITLE
Add issue update workflows

### PR DESCRIPTION
**Pull request by RickWinter**
Enable the GitHub issue update workflows. I took these directly from the Azure/azure-sdk-for-go.git repo

### CODE REVIEW

File: .github/workflows/event-processor.yml
Comment (RickWinter) [inline .github/workflows/event-processor.yml:154]:
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          
```

File: .github/workflows/event.yml
Comment (RickWinter) [inline .github/workflows/event.yml:25]:
```suggestion
          token: ${{ secrets.GITHUB_TOKEN }}
          
```

File: .github/workflows/scheduled-event-processor.yml
Comment (RickWinter) [inline .github/workflows/scheduled-event-processor.yml:144]:
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          
```

**Review by heaths — APPROVED** [review]
✅ No Easter eggs

**Comment by weshaggard** [comment]
@JimSuplizio @hallipr should we add this to the sync list?

File: .github/workflows/event-processor.yml
Comment (weshaggard) [inline .github/workflows/event-processor.yml:38]: We will need to setup these secrets before this action will work.

File: .github/workflows/event-processor.yml
Comment (weshaggard) [inline .github/workflows/event-processor.yml:51]: We will need to sync the labels to this repo before some of these processes will work.

Comment (heaths) [reply .github/workflows/event-processor.yml:51]: The repositories.txt file I updated a long time ago. Or are you talking about a different sync?

Comment (weshaggard) [reply .github/workflows/event-processor.yml:51]: I just want to be sure the labels are in-sync

**Comment by JimSuplizio** [comment]
> @JimSuplizio @hallipr should we add this to the sync list?
@weshaggard and @RickWinter 

I have a concern with this. We push the entire .github/workflows directory and, for other languages, they only contain the same 3 yml files that are in this PR but Rust's [.github/workflows directory](https://github.com/Azure/azure-sdk-for-rust/tree/main/.github/workflows) already contains other workflow yml files.

**Review by JimSuplizio — COMMENTED** [review]
@RickWinter, you're missing a file. There needs to be an [event-processor.config]([event-processor.config](https://github.com/Azure/azure-sdk-for-java/blob/main/.github/event-processor.config) file in the .github directory (not .github/workflows). The file I'd linked is from Java, but pretty much all of the language repositories use the same config (everything is turned on). With the config file, the event processor and scheduled event processor won't run.

CC: @weshaggard 
@RickWinter, there's one other thing. If you're going to be running events, then we're also going to need to setup azure-sdk-for-rust to run the codeowners-linter.
[UPDATE] @hallipr already has a [PR](https://github.com/Azure/azure-sdk-tools/pull/8737/files#diff-b5d7243f40e5fa06023749f464cfd8b8b32485140592e31036ceb7a2b6a2f7fdR22) adding Rust to the eng/common sync. I'll go ahead and create the CODEOWNERS_baseline_errors.txt for Rust, if there are any errors.

File: .github/workflows/event-processor.yml
Comment (JimSuplizio) [reply .github/workflows/event-processor.yml:51]: @heaths and @weshaggard, every repo in repositories.txt has its repository labels in the repo/labels store which is used by both the linter and event processor. The other piece of this is the AI label service, which is invoked during the initial issue triage rule. I've already chatted with @jsquire this morning and azure-sdk-for-rust is setup in the AI label service.

**Comment by weshaggard** [comment]
@RickWinter is there anything pressing us to get this in currently? I would prefer we wait until the track2 stuff gets moved into main and we can enable the general syncing.

**Comment by heaths** [comment]
I want to make sure the CIs and all checks are still running for PRs and that future PRs aren't held up. We can't lose coverage.

File: .github/workflows/event-processor.yml
Comment (weshaggard) [reply .github/workflows/event-processor.yml:38]: We will need to wait until @benbp is back as he didn't grant me owner to the app registration we are using here. https://ms.portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/~/Owners/appId/a6dd2dfe-7352-41a7-9020-05301c3bca1a/isMSAApp~/false

Comment (benbp) [reply .github/workflows/event-processor.yml:38]: Added @weshaggard and @mikeharder as owner on this.

**Comment by hallipr** [comment]
> [UPDATE] @hallipr already has a [PR](https://github.com/Azure/azure-sdk-tools/pull/8737/files#diff-b5d7243f40e5fa06023749f464cfd8b8b32485140592e31036ceb7a2b6a2f7fdR22) adding Rust to the eng/common sync.

That PR is pending merge of the current `feature/track2` branch to main.  We don't currently have eng/common in main

**Comment by heaths** [comment]
`main` branch has been moved to `legacy`. I changed the base here, but should this actually target `main` after being rebased (since I'm going to merge `feature/track2` into `main`)?

/cc @RickWinter @hallipr @weshaggard

**Comment by RickWinter** [comment]
> `main` branch has been moved to `legacy`. I changed the base here, but should this actually target `main` after being rebased (since I'm going to merge `feature/track2` into `main`)?
> 
> /cc @RickWinter @hallipr @weshaggard

My expectation is this would go into main once main is populated with feature/track2

**Comment by heaths** [comment]
> My expectation is this would go into main once main is populated with feature/track2

The context of my question is that the EngSys team has all sorts of automated processes. I don't know if manually checking these files in is still the correct thing to do, or if they have some sync script or whatnot. @weshaggard?

**Comment by weshaggard** [comment]
Yes we do have a sync process for these and thus should close this PR. @hallipr can you compete your PR https://github.com/Azure/azure-sdk-tools/pull/8737/files and do a sync?

**Pull request closed without merging**
